test/services/katello/pulp/repository/docker_test.rb

```diff
@@ -42,6 +44,19 @@ def test_create
           delete_repo(@repo)
         end
 
+        def test_index_content
+          @ssh.root.mirror_on_sync = true
+
+          service = Katello::Pulp::Repository::Docker.new(@ssh, @master)
+          service.create
+          RepositorySupport.create_and_sync_repo(@ssh)
+
+          @ssh.index_content
+          assert_equal @ssh.docker_tags.count, 2
+        ensure
+          delete_repo(@ssh)
+        end
+
```

Comment:
I'm not sure this test actually reproduces the problem?  I think you'd have to then create some content_view repo as well, and then index, passing in the source repo.  Although there'd be nothing inthe content view repo.  Maybe you could copy units over to the cv repo prior

Reply:
Ok, I'll bolster this a bit

Reply:
I've updated the test.